### DISCUSSION TITLE
HEVC/h265 Support

### DISCUSSION THREAD

**Discussion by artem-alexandrov** [nwjs-ffmpeg-prebuilt/nwjs-ffmpeg-prebuilt#158] [Q&A]
Does it support h265? One of users claim that it doesn't, is it possible to add h265 support to the lib?

**Comment by zkrige** [comment]
I'd love to know how to add support for more codecs to the library.

I took over maintenance of this repo a while back, but I'm not particularly familiar with ffmpeg specifics.

If someone can figure out how to add support for more codecs, that would be great

**Comment by nebular** [comment]
to add support for more codecs you typically add compilation flags and install the codec requirements. For example, for H265 / HEVC: https://trac.ffmpeg.org/wiki/Encode/H.265

**Comment by zkrige** [comment]
I'm not quite sure how to get ffmpeg compiler flags into this piece of code

 ```
if (program.arch === 'ia32') {
        await execAsync('gn', 'gen', 'out/Default', '--args="chrome_pgo_phase=0 is_debug=false enable_nacl=false is_component_ffmpeg=true proprietary_codecs=true is_official_build=true target_cpu=\\"x86\\" ffmpeg_branding=\\"Chrome\\""');
    } else if (program.arch === 'x64') {
        await execAsync('gn', 'gen', 'out/Default', '--args="chrome_pgo_phase=0 is_debug=false enable_nacl=false is_component_ffmpeg=true proprietary_codecs=true is_official_build=true target_cpu=\\"x64\\" ffmpeg_branding=\\"Chrome\\""');
    } else if (program.arch === 'arm') {
        await execAsync('gn', 'gen', 'out/Default', '--args="chrome_pgo_phase=0 is_debug=false enable_nacl=false is_component_ffmpeg=true proprietary_codecs=true is_official_build=true target_cpu=\\"arm\\" ffmpeg_branding=\\"Chrome\\""');
    } else if (program.arch === 'arm64') {
        await execAsync('gn', 'gen', 'out/Default', '--args="chrome_pgo_phase=0 is_debug=false enable_nacl=false is_component_ffmpeg=true proprietary_codecs=true is_official_build=true target_cpu=\\"arm64\\" ffmpeg_branding=\\"Chrome\\""');
    }
```

**Comment by nebular** [comment]
I also don´t know how to propagate FFMPEG build options from the Chrome build.
Looking at the source code, stuff is at ```build/chromium/src/third_party/ffmpeg```.

I tried to run ```./configure --enable-gpl --enable-libx265``` from that directory and it worked, but no idea how to incorporate this in the ninja files.
